src/legacy/core_plugins/data/public/filter/filter_manager/filter_manager.ts

```diff
@@ -85,9 +85,17 @@ export class FilterManager {
   private handleStateUpdate(newFilters: Filter[]) {
     // global filters should always be first
     newFilters.sort((a: Filter, b: Filter): number => {
-      if (a.$state && a.$state.store === FilterStateStore.GLOBAL_STATE) {
+      // TypeScript, this shouldn't happen
+      if (!a.$state || !b.$state) return 0;
+      if (
+        a.$state.store === FilterStateStore.GLOBAL_STATE &&
+        b.$state.store !== FilterStateStore.GLOBAL_STATE
+      ) {
         return -1;
-      } else if (b.$state && b.$state.store === FilterStateStore.GLOBAL_STATE) {
+      } else if (
+        a.$state.store !== FilterStateStore.GLOBAL_STATE &&
+        b.$state.store === FilterStateStore.GLOBAL_STATE
+      ) {
```

Comment:
It looks like you can simplify this function to a one-liner:

![image](https://user-images.githubusercontent.com/9773803/60720984-bc6abd00-9f2c-11e9-9889-1af4eb14af00.png)

Reply:
This won't return 0 in case they have the same store (which was the  problem with the original logic).
But I'll definitely use the `$state!` notation